homeassistant/components/geocaching/sensor.py

```diff
@@ -25,58 +32,135 @@ class GeocachingSensorEntityDescription(SensorEntityDescription):
         entity_registry_visible_default=False,
         value_fn=lambda status: status.user.hide_count,
     ),
     GeocachingSensorEntityDescription(
         key="favorite_points",
         translation_key="favorite_points",
-        native_unit_of_measurement="points",
         entity_registry_visible_default=False,
         value_fn=lambda status: status.user.favorite_points,
     ),
     GeocachingSensorEntityDescription(
         key="souvenir_count",
         translation_key="souvenir_count",
-        native_unit_of_measurement="souvenirs",
         value_fn=lambda status: status.user.souvenir_count,
     ),
     GeocachingSensorEntityDescription(
         key="awarded_favorite_points",
         translation_key="awarded_favorite_points",
-        native_unit_of_measurement="points",
         entity_registry_visible_default=False,
         value_fn=lambda status: status.user.awarded_favorite_points,
     ),
 )
 
 
+@dataclass(frozen=True, kw_only=True)
+class GeocachingCacheSensorDescription(SensorEntityDescription):
+    """Define Sensor entity description class."""
+
+    value_fn: Callable[[GeocachingCache], StateType | datetime.date]
+
+
+CACHE_SENSORS: tuple[GeocachingCacheSensorDescription, ...] = (
+    GeocachingCacheSensorDescription(
+        key="found_date",
+        device_class=SensorDeviceClass.DATE,
+        value_fn=lambda cache: cache.found_date_time,
+    ),
+    GeocachingCacheSensorDescription(
+        key="favorite_points",
+        value_fn=lambda cache: cache.favorite_points,
+    ),
+    GeocachingCacheSensorDescription(
+        key="hidden_date",
+        device_class=SensorDeviceClass.DATE,
+        value_fn=lambda cache: cache.hidden_date,
+    ),
+)
+
+
 async def async_setup_entry(
     hass: HomeAssistant,
     entry: GeocachingConfigEntry,
     async_add_entities: AddConfigEntryEntitiesCallback,
 ) -> None:
     """Set up a Geocaching sensor entry."""
     coordinator = entry.runtime_data
-    async_add_entities(
-        GeocachingSensor(coordinator, description) for description in SENSORS
+
+    entities: list[Entity] = []
+
+    entities.extend(
+        GeocachingProfileSensor(coordinator, description)
+        for description in PROFILE_SENSORS
     )
 
+    status = coordinator.data
+
+    # Add entities for tracked caches
+    entities.extend(
+        GeoEntityCacheSensorEntity(coordinator, cache, description)
+        for cache in status.tracked_caches
+        for description in CACHE_SENSORS
+    )
+
+    async_add_entities(entities)
+
+
+# Base class for a cache entity.
+# Sets the device, ID and translation settings to correctly group the entity to the correct cache device and give it the correct name.
+class GeoEntityBaseCache(GeocachingCacheEntity, SensorEntity):
+    """Base class for cache entities."""
+
+    cache: GeocachingCache
+
+    def __init__(
+        self,
+        coordinator: GeocachingDataUpdateCoordinator,
+        cache: GeocachingCache,
+        entity_type: str,
```

Comment:
```suggestion
        key: str,
```
Key is a much more common name for this

Reply:
Fixed in 10ae69a76ff